daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -699,7 +699,14 @@ object NodeInfo extends Enum {
     protected sealed trait BooleanKind extends AnySimpleType.Kind
     case object Boolean extends PrimTypeNode(AnyAtomic) with BooleanKind with PrimNonNumeric with BooleanView {
       type Kind = BooleanKind
-      protected override def fromString(s: String): DataValueBool = s.toBoolean
+      protected override def fromString(s: String): DataValueBool = {
+        // Note: Any boolean default value other than true, false, 1, or 0
+        // will throw a SDE resulting from a SAXParseException
+        s match {
+          case "0" | "false" => false
+          case "1" | "true" => true
+         }
```

Comment:
Actually ``fromXMLString`` and thus ``fromString`` are used in many places other that defaulting, so that comment isn't always going to be accurate. And in some of those cases, we can't rely on schema validation/SAX errors. For example, you could have a schema like this:
```xml
<xs:element name="root">
  <xs:complexType>
    <xs:sequence>
      <xs:element name="str" type="xs:string" ... />
      <xs:element name="bool" type="xs:boolean" dfdl:inputValueCalc="{ xs:boolean(../str) }" />
    </xs:string>
  </xs:complexType>
</xs:element>
```
In this case, the conversion of str to an xs:boolean will use this fromString function and isn't validated by SAX. In that case, we robably would get a scala MatchError, which wouldn't be good for diagnostics.

So I would suggest removing that comment and having a default case that's something like:

```scala
case _ => throws an IllegalArgumentException("Must be one of 0, 1, true, or false")
```
Then ``fromXMLString`` will catch that and create the appropriate error when it does occur.

If we don't have a test like the above, it might also be worth adding to make sure inputCalueCalc runtime conversions give an appropriate error.

Reply:
Tests and header added and all changes pushed